pkg/logql/rangemapper_test.go

```diff
@@ -1750,4 +1751,18 @@ func Test_FailQuery(t *testing.T) {
 	require.Error(t, err)
 	_, _, err = rvm.Parse(`topk(0, sum(count_over_time({app="foo"} | json |  __error__="" [15m])))`)
 	require.Error(t, err)
+
+	// Check fixes for bug where missing or empty parameters for regexp and pattern parsers threw a panic
+	// Missing parameter to regexp parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | regexp [1d])))`)
+	require.ErrorIs(t, err, logqlmodel.ErrParse)
+	// Empty parameter to regexp parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | regexp \"\" [1d])))`)
+	require.ErrorIs(t, err, logqlmodel.ErrParse)
+	// Empty parameter to pattern parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | pattern \"\" [1d])))`)
+	require.ErrorIs(t, err, logqlmodel.ErrParse)
+	// Empty parameter to json parser
+	_, _, err = rvm.Parse(`topk(10,sum by(namespace)(count_over_time({application="nginx", site!="eu-west-1-dev"} |= "/artifactory/" != "api" != "binarystore" | json [1d])))`)
```

Comment:
Why should this one fail?

Reply:
I assume you're referring to the pattern test. The pattern parser throws an error if there are no captures: see https://github.com/grafana/loki/blob/main/pkg/logql/log/pattern/ast.go#L20.

Reply:
I'm referring to the json test

Reply:
That was added to make sure I didn't mess up the other stages. Note that `rvm.Parse` doesn't return an error there.

Reply:
Ah, I see. Thanks!